app/src/main/java/com/example/faunaapp/model/AddEntryModel.java

```diff
@@ -0,0 +1,106 @@
+package com.example.faunaapp.model;
+
+import android.os.AsyncTask;
+import android.util.Log;
+
+import com.apollographql.apollo.ApolloCall;import com.apollographql.apollo.ApolloClient;
+import com.apollographql.apollo.api.Response;
+import com.apollographql.apollo.exception.ApolloException;
+import com.apollographql.apollo.request.RequestHeaders;
+import com.example.faunaapp.DTO.Entry;
+import com.example.faunaapp.Helper.EntriesAdapter;
+import com.example.faunaapp.client.Client;
+import com.faunaapp.graphql.CreateTaskMutation;
+import com.faunaapp.graphql.type.Category;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+import java.util.TimeZone;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import kotlin.text.Regex;
+
+public class AddEntryModel implements IAddEntryModel {
+
+    private static ApolloClient apolloClient;
+    private CreateTaskAsync createTaskAsync;
+
+    public AddEntryModel(Client client) {
+        apolloClient = client.getClient();
+        createTaskAsync = new CreateTaskAsync();
+    }
+
+
+    @Override
+    public void submit(Entry entry) {
+      createTaskAsync.doInBackground(getISO8601Format(entry.getDate(), entry.getTime()), entry.getHeading(),entry.getTitle(),Category.MEDICINE.toString());
+    }
+
+    private String getISO8601Format(String date, String time) {
+        StringBuilder dateTimeBuilder = new StringBuilder();
```

Comment:
I think you should find some library methods to help you with this, instead of using a StringBuilder.
As you will find out working with date time is really hard, and you will not be able to take care of it on your own.
I recommend seeing the following video about it. 😄  https://www.youtube.com/watch?v=-5wpm-gesOY

Reply:
oki

Reply:
But how do I get more data than date(year, month, day) and time (hours, minutes) cause I need more to create an ISO8601 format

Reply:
I have the following, which has some error I have not correct yet, where the backend does not accept the resulting string yet. But you could use this for inspiration I guess
`ZonedDateTime.of(year, month + 1, day, 0, 0,0,0, ZoneId.of( "Europe/Copenhagen" ))`

Reply:
I then use `.format( DateTimeFormatter.ISO_OFFSET_DATE_TIME )` method on ZonedDateTime.